orttraining/orttraining/test/training_ops/cuda/softmax_test.cc

```diff
@@ -44,8 +44,13 @@ static void TestSoftmaxGrad(const std::vector<int64_t>& dY_dims,
 
   // create rand inputs
   RandomValueGenerator random{};
-  std::vector<float> dY_data = random.Uniform<float>(dY_dims, -10.0f, 10.0f);
-  std::vector<float> Y_data = random.Uniform<float>(Y_dims, -10.0f, 10.0f);
+  std::vector<float> dY_data = random.Uniform<float>(dY_dims, 0.0f, 1.0f);
+  std::vector<float> Y_data = random.Uniform<float>(Y_dims, 0.0f, 1.0f);
+  // Add 1e-2 for numerical stability
+  for (int index = 0; index < Y_data.size(); index += 1) {
+    Y_data[index] += 1e-2f;
+  }
```

Comment:
Great investigation!!

FYI, these few line can also be written in
std::vector<float> Y_data = random.Uniform<float>(Y_dims, 0.02f, 1.02f);

Also, a even more realistic test data is to ensure Sum(Yi) = 1, where Yi is in [0, 1], since Y is the output of softmax. But this property of Y is not needed to ensure the numerical stabilyt, thus also skipped by pytorch test.

Reply:
Thanks!